halo2_debug/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+use rand_core::block::BlockRng;
+use rand_core::block::BlockRngCore;
+use rand_core::OsRng;
+use tiny_keccak::Hasher;
+
+// One number generator, that can be used as a deterministic Rng, outputing fixed values.
+pub struct OneNg {}
+
+impl BlockRngCore for OneNg {
+    type Item = u32;
+    type Results = [u32; 16];
+
+    fn generate(&mut self, results: &mut Self::Results) {
+        for elem in results.iter_mut() {
+            *elem = 1;
+        }
+    }
+}
+
+pub fn one_rng() -> BlockRng<OneNg> {
+    BlockRng::<OneNg>::new(OneNg {})
+}
+
+// Random number generator for testing
+
+pub fn test_rng() -> OsRng {
+    OsRng
+}
```

Comment:
I think we should use seedable pseudo random generators for tests.  Is the plan to change this in the future?  Is this related to the need for the rng to be clonable where each clone generates different randomness, as required by some parallel code?

In fact what confuses me is that the PR is called "Deterministic tests" but most of the tests call `test_rng` which is not deterministic.